plugins/module_utils/system_access_users_utils.py

```diff
@@ -39,70 +39,114 @@
+    Exception raised when the return value of the instance is not what is expected
     """
 
 
-class OPNSenseNotValidBase64APIKeyError(Exception):
+class OPNSenseGroupNotFoundError(Exception):
     """
-    Exception raised when a not valid base32 api code is provided
+    Exception raised when an OPNsense group is not found.
     """
 
 
-class OPNSenseCryptReturnError(Exception):
+class OPNSenseHashVerifyReturnError(Exception):
     """
     Exception raised when the return value of the instance is not what is expected
     """
 
 
-class OPNSensePasswordVerifyReturnError(Exception):
+class OPNSenseNotValidBase64APIKeyError(Exception):
     """
-    Exception raised when the return value of the instance is not what is expected
+    Exception raised when a not valid base32 api code is provided
     """
 
 
-def password_verify(existing_user_password: str, password: Optional[str]) -> bool:
+def hash_verify(existing_hashed_string: str, plain_string: Optional[str]) -> bool:
     """
-    Verify if provided password matches the stored password using OPNsense's PHP command.
+    Verifies if a plain string matches an existing hashed string.
 
     Args:
-        existing_user_password (str): The hashed password stored in the XML config.
-        password (str): The plaintext password to verify.
+        existing_hashed_string (str): The existing hashed string to verify against.
+        plain_string (Optional[str]): The plain string to verify.
 
     Returns:
-        bool: True if passwords match, False otherwise.
+        bool: True if the plain string matches the hashed string, otherwise False.
 
     Raises:
-        OPNSensePasswordVerifyReturnError: If an error occurs during verification.
+        OPNSenseHashVerifyReturnError: If an error occurs during hash verification.
     """
 
-    if password is None:
+    if plain_string is None:
         return False
 
-    # check if current password matches hash
-    password_matches = opnsense_utils.run_command(
+    # escape plain_string
+    escaped_string = plain_string.replace("\\", "\\\\").replace("'", "\\'")
+
+    # check if current plain_string matches hash
+    hash_matches = opnsense_utils.run_command(
         php_requirements=[],
-        command=f"password_verify('{password}','{existing_user_password}');",
+        command=f"var_dump(password_verify('{escaped_string}','{existing_hashed_string}'));",
     )
 
-    if password_matches.get("stderr"):
-        raise OPNSensePasswordVerifyReturnError("error encounterd verifying password")
+    if hash_matches.get("stderr"):
+        raise OPNSenseHashVerifyReturnError(
+            f"error encounterd verifying hash {hash_matches.get('stderr')}"
+        )
 
-    # if return code of password_matches not equals 1, it's a match
-    if password_matches.get("stdout") != "1":
+    # if return code of hash_matches is true, it's a match
+    if hash_matches.get("stdout") == "bool(true)":
         return True
 
     return False
 
```

Comment:
The boolean value can be returned directly
```suggestion
    return hash_matches.get("stdout") == "bool(true)"
```

Reply:
Thanks for the suggestion, fixed in ea1b1d0.